mocks/firestore.js

```diff
@@ -219,9 +219,59 @@ FakeFirestore.DocumentReference = class {
 
   set(object) {
     mockSet(...arguments);
+    this._updateData(object);
```

Comment:
I added a PR that could add a new `mutable` flag to the options hash we recently introduced.

Can you just wrap this in an `if`

```
if (this.options.mutable) {
  this._updateData(object);
}
```

Reply:
Yes, of course, I'll make these changes.  Would you be OK if this was a `throw` though?

For me personally, it would make me sleep a lot easier to to have that little extra check that I haven't got any accidental mutations in something I expect to be a read-only path.

Reply:
I'm not sure I follow. What would be throwing the error? I agree that I want users who are on the read-only path to not have any headaches/mysteries.

Reply:
MockFirestore would throw the error rather than silently failing.  Might make usage a bit clearer/easier.

nvm - I'll leave it as-is to finish this off.  My use-case is pretty niche and I can achieve it without introducing unwanted side-effects here!